pkgs/cronet_http/lib/src/cronet_client.dart

```diff
@@ -186,26 +212,114 @@ class CronetClient extends BaseClient {
           'HTTP request failed. Client is already closed.', request.url);
     }
 
-    if (_engine == null) {
-      // Create the future here rather than in the [fromCronetEngineFuture]
-      // factory so that [close] does not have to await the future just to
-      // close it in the case where [send] is never called.
-      //
-      // Assign to _engineFuture instead of just `await`ing the result of
-      // `CronetEngine.build()` to prevent concurrent executions of `send`
-      // from creating multiple [CronetEngine]s.
-      _engineFuture ??= CronetEngine.build();
-      try {
-        _engine = await _engineFuture;
-      } catch (e) {
-        throw ClientException(
-            'Exception building CronetEngine: ${e.toString()}', request.url);
-      }
-    }
+    _engine ??= CronetEngine.build();
 
     final stream = request.finalize();
-
     final body = await stream.toBytes();
+    final responseCompleter = Completer<StreamedResponse>();
```

Comment:
[nit] This method is going from ~100 lines to ~140 lines.

Are there any blocks that can be pulled out into other methods that we compose here?

Reply:
I'll try moving `cronetCallbacks` to a separate class in a follow-up PR and see how it looks